frontend_tests/node_tests/pm_list_data.js

```diff
@@ -0,0 +1,240 @@
+    list_info = get_list_info();
+
+    assert.deepEqual(list_info.convos_to_be_shown.length, 8);
+    assert.deepEqual(list_info.more_convos_unread_count, 0);
+
+    // Now after adding one more PM, the length in default view would be 9 which would
+    // exceed `max_convos_to_show_with_unreads`, so the number of unreads present in
+    // `more conversations` li-item should be 1 and also the length of `convos_to_be_shown` should be 8.
+    pm_conversations.recent.insert([107], timestamp);
+    list_info = get_list_info();
+    assert.deepEqual(list_info.convos_to_be_shown.length, 8);
+    assert.deepEqual(list_info.more_convos_unread_count, 1);
+
+    // Now, here we test the functionality when the list is zoomed, now according to the logic
+    // we should have all the 9 conversations visible to us.
+    const zoomed = true;
+    list_info = get_list_info(zoomed);
+    assert.deepEqual(list_info.convos_to_be_shown.length, 9);
+
+    // Now, we would like to test the functionality of active_pm being pushed into the `convos_to_be_shown`
+    // even when the length of `convos_to_be_shown` reaches it's maximum value of `max_convos_to_be_shown`
+    // so we added 6 conversations to pm_conversations with 0 unread in all of them, and now we made the
+    // first conversation as active, so now according to the logic -- our default view should display 6
+    // conversations among which the last one should have `is_active` as true.
+    override(unread, "num_unread_for_person", () => 0);
+    pm_conversations.clear_for_testing();
+    pm_conversations.recent.insert([101], timestamp);
+    pm_conversations.recent.insert([102], timestamp);
+    pm_conversations.recent.insert([103], timestamp);
+    pm_conversations.recent.insert([104], timestamp);
+    pm_conversations.recent.insert([105], timestamp);
+    pm_conversations.recent.insert([106], timestamp);
+
+    // currently we have 6 conversations in total, but only recent 5 would be visible.
+    list_info = get_list_info();
+    assert.deepEqual(list_info.convos_to_be_shown.length, 5);
+
+    // now, after setting the 6th one as active, we would have all 6 as visible with the last one's `is_active` as true.
+    function set_filter(operators) {
+        operators = operators.map((op) => ({
+            operator: op[0],
+            operand: op[1],
+        }));
+        narrow_state.set_current_filter(new Filter(operators));
+    }
+    set_filter([["pm-with", "alice@zulip.com"]]);
+    list_info = get_list_info();
+    assert.deepEqual(list_info.convos_to_be_shown.length, 6);
+    assert.deepEqual(list_info.convos_to_be_shown[5], {
+        recipients: "Alice",
+        user_ids_string: "101",
+        unread: 0,
+        is_zero: true,
+        is_active: true,
+        url: "#narrow/pm-with/101-alice",
+        status_emoji_info: {emoji_code: 20},
+        user_circle_class: "user_circle_empty",
+        is_group: false,
+    });
```

Comment:
Ideally the testing here should verify the ordering of the conversations is correct, not just the length.  We do this nicely in the `topic_list_data` tests, like this:

```
    assert.deepEqual(                                                                                    
        list_info.items.map((li) => li.topic_name),                                                      
        ["topic 0", "topic 1", "topic 2", "topic 3", "topic 5", "topic 6", "topic 7", "topic 8"],        
    );                                                                                                   
```

Reply:
Hmm, Understood;

Made the changes, Updated the PR;
thanks.